bed/compare.go

```diff
@@ -159,6 +159,24 @@ func CompareChromEndByChrom(a Bed, b Bed) int {
 	}
 	return CompareChromEnd(a, b)
 }
+//CompareDistance compares beds by chromStart and chromEnd and returns the minimum distance between the two beds.
+func CompareDistance(a Bed, b Bed) int {
+	chromComp := strings.Compare(a.Chrom, b.Chrom)
+	if chromComp != 0 {
+		return chromComp  //maybe this should flag to tell the user the chrom are not the same, otherwise it just gives back a "-1" or "1".
+		// This also a potential problem for "func compare" starting at line 110 above.
+	}
+	if Overlap (a, b) {
+		return 0
+	}
+	if a.ChromStart - b.ChromEnd > 0 { //only positive if bed "a" is downstream of "b" bed.
+		return int(a.ChromStart - b.ChromEnd)
+	}
+	if b.ChromStart - a.ChromEnd > 0 { //only positive if bed "b" is downstream of "a" bed.
+		return int(b.ChromStart - a.ChromEnd)
+	}
+	return ??? // I dont know what to return here?
```

Comment:
It's maybe not ideal, but one way to handle this is to return an integer where -1 is always 'different chromosomes' and any value >=0 is the inter-chromosomal distance. You could leave the code as-is except changing line 166 to `return -1`.

Reply:
Also, for the problem of what to return at the end. It seems like all valid bed entries should be true for one of the above if statements so it you have not returned by line 178, something bad has happened and there is likely a bug in the code calling this function. One way to handle this is:

```
// rest of the function //

log.Panic("something went wrong")
return 0 // unreachable
```

Reply:
Ahh, great point! So I would just put the "log.Panic..." in place of the return or after the return? I think i like this more than the "-1" if this is something that would replace it. @dasnellings

Reply:
The log.Panic will kill the program so the final return is never actually used, it is only there to satisfy the compile requirement that a function with a returned value ends in a return. I put zero, but it can be anything.